src/mainboard/pcengines/apu2/mainboard.c

```diff
@@ -262,15 +280,16 @@ static int mainboard_smbios_type17(DMI_INFO *agesa_dmi, int *handle,
 				agesa_dmi->T17[0][0][0].BankLocator);
 	t->memory_type = agesa_dmi->T17[0][0][0].MemoryType;
 	t->type_detail = *(u16 *)&agesa_dmi->T17[0][0][0].TypeDetail;
-	t->speed = agesa_dmi->T17[0][0][0].Speed;
+	t->speed = ddr_speed_from_bus_speed(agesa_dmi->T17[0][0][0].Speed);
 	t->manufacturer = agesa_dmi->T17[0][0][0].ManufacturerIdCode;
 	t->serial_number = smbios_add_string(t->eos,
 				agesa_dmi->T17[0][0][0].SerialNumber);
 	t->part_number = smbios_add_string(t->eos,
 				agesa_dmi->T17[0][0][0].PartNumber);
 	t->attributes = agesa_dmi->T17[0][0][0].Attributes;
 	t->extended_size = agesa_dmi->T17[0][0][0].ExtSize;
-	t->clock_speed = agesa_dmi->T17[0][0][0].ConfigSpeed;
+	t->clock_speed = ddr_speed_from_bus_speed(
```

Comment:
See this field is a clock speed and it should remain unchanged? What is the value of agesa_dmi->T17[0][0][0].ConfigSpeed? Section 7.18:
https://www.dmtf.org/sites/default/files/standards/documents/DSP0134_3.1.1.pdf

Reply:
Both of these values were 2x bigger before version v4.12.0.1. (You can see it in files attached to issue https://github.com/pcengines/apu2-documentation/issues/176 )
and as far as I understand datasheet both of these values are speed of the memory and are given in MT/s. Only the speed of the bus which would be in MHz should remain unchanged.

Reply:
Yes, becasue the were manually set.  The ConfigSpeed should represent the configured memory clock on the memory bus, so here the conversion is not desired.